dockstore-webservice/src/main/java/io/dockstore/webservice/resources/MetadataResource.java

```diff
@@ -287,6 +287,9 @@ public Response getRunnerDependencies(
             return Response.noContent().build();
         }
         boolean unwrap = !("json").equals(output);
+        if (!PipHelper.validateSemVer(clientVersion)) {
+            throw new CustomWebApplicationException("Invalid client version: " + clientVersion + ". Must match pattern: " + PipHelper.SEM_VER_PATTERN.pattern(), HttpStatus.SC_BAD_REQUEST);
```

Comment:
I'm not sure our all our users will know regexes (and I often have to refresh myself if it's a non-trivial expression), so a more user-friendly message might be nice.